packages/tables/backend/sirius-components-tables/src/main/java/org/eclipse/sirius/components/tables/descriptions/ColumnDescription.java

```diff
@@ -0,0 +1,171 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.components.tables.descriptions;
+
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.eclipse.sirius.components.annotations.Immutable;
+import org.eclipse.sirius.components.representations.VariableManager;
+
+/**
+ *
+ * The description of the column.
+ *
+ * @author arichard
+ */
+@Immutable
+public final class ColumnDescription {
+
+    private UUID id;
+
+    private Function<VariableManager, String> labelProvider;
+
+    private Function<VariableManager, String> featureNameProvider;
+
+    private BiFunction<VariableManager, String, String> cellTypeProvider;
```

Comment:
Why a cell type provider? Just compute the proper cell like in `org.eclipse.sirius.components.diagrams.description.NodeDescription#styleProvider` which can return an Ellipse, a Rectangle, a List, etc...